internal/compiler-bridge/src/main/scala/xsbt/API.scala

```diff
@@ -118,7 +119,11 @@ final class API(val global: CallbackGlobal) extends Compat with GlobalHelpers wi
         if (!symbol.isLocalClass) {
           val classFileName = s"${names.binaryName}.class"
           val outputDir = global.settings.outputDirs.outputDirFor(sourceFile).file
-          val classFile = new java.io.File(outputDir, classFileName)
+          val classFile = if (STJ.enabled) {
+            new java.io.File(STJ.JaredClass(outputDir, classFileName))
```

Comment:
It feels like it would significantly improve type safety to use an enum to represent the two cases. But I could also imagine that that would be an even larger refactoring than this already is... so perhaps better as a followup.